internal/configs/virtualserver_test.go

```diff
@@ -1275,6 +1275,119 @@ func TestGenerateVirtualServerConfigWithSpiffeCerts(t *testing.T) {
 	}
 }
 
+func TestGenerateVirtualServerConfigWithInternalRoutes(t *testing.T) {
+	t.Parallel()
+	virtualServerEx := VirtualServerEx{
+		VirtualServer: &conf_v1.VirtualServer{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name:        "cafe",
+				Namespace:   "default",
+				Annotations: map[string]string{"nsm.nginx.com/internal-route": "true"},
```

Comment:
@ciarams87 Do we want an annotation on a CRD resource?
Or do we want a property?
I realize an annotation is probably faster to implement, since Ingress uses an annotation.
But annotation is an anti-pattern to VS, VSR, TS, etc.  Isn't it?

Reply:
Sorry, I didn't notice this comment until now! 

Using an annotation to enable a feature is generally an anti-pattern for our CRDs, however it is for an internal only use case with nothing customer facing (I am making an assumption here, so please correct me if I am wrong!) so in this case I think an annotation is OK. We may not want to modify the CRD for an internal only flag - but it's not something we have encountered before.

Reply:
Oh, ok I didn't realise this - In that case, yes this is an anti-pattern, and we should extend the VS CRD instead to add this as a field/ property to the definition. I'm happy to help if you need anything!

Reply:
I pushed an update that uses a CRD field instead of an annotation. Let me know if any other changes are necessary.